### DISCUSSION TITLE
Add whatsapp support for group admin deletion of messages sent by others

### DISCUSSION THREAD

**Discussion by leonardo-fabricio** [d99kris/nchat#145] [Ideas]
Hi, is there any way to delete the message for everyone in your library? I couldn't find anything that helped me.

Thank you for the answer.

**Comment by d99kris** [comment]
Hi @leonardo-fabricio - I believe nchat attempts to revoke the message upon delete, as far as the underlying messaging platform permits. I tested just now, in a normal (1-to-1) chat a message sent and subsequently deleted gets:
- Fully removed from the chat for the other user in Telegram
- Replaced with "This message was deleted" for the other user in WhatsApp

I believe this should be the same as what happens if you "Delete for everyone" (WhatsApp) or use "Also delete for X" (Telegram) in the official mobile apps. 

In order words, nchat always attempts to delete a message for everyone. Let me know if you're observing something else?

**Edit:** And to clarify, the way to do this in nchat is to use arrow up/down to select a message and then press ctrl-d (and press enter to confirm).

**Reply by ElielsonU** [reply]
Hi! I,m with the same issue. I believe he's an admin trying to delete messages from other users. An admin can do this in this package? Or do you know some way to do this?

**Reply by d99kris** [reply]
I'll need to look into it. Btw, is this on Telegram or WhatsApp?

**Reply by ElielsonU** [reply]
In WhatsApp. I'm using the whatsmeow lib, and they have this function in the lib

**Reply by d99kris** [reply]
Thanks for clarifying @ElielsonU. So deleting messages sent by others as a group admin on whatsapp is currently not supported in nchat. Like you mentioned it looks like whatsmeow supports it. I'll look into adding support for it and update here again.

**Reply by d99kris** [reply]
I've now added whatsapp support for group admin deletion of messages sent by others. Please let me know if any issues are encountered. Thanks!

**Reply by d99kris** [reply]
I will proceed to close this topic, feel free to re-open or initiate a new discussion if you encounter any other issues.
